include/swift/ABI/TaskLocal.h

```diff
@@ -44,6 +44,14 @@ class TaskLocal {
     /// lookups by skipping empty parent tasks during get(), and explained
     /// in depth in `createParentLink`.
     IsParent = 0b01,
+    /// The task local binding was created inside the body of a `withTaskGroup`,
+    /// and therefore must either copy it, or crash when a child task is created
+    /// using 'group.addTask' and it would refer to this task local.
+    ///
+    /// Items of this kind must be copied by a group child task for access
+    /// safety reasons, as otherwise the pop would happen before the child task
+    /// has completed.
+    IsNextCreatedInTaskGroupBody = 0b10,
   };
```

Comment:
You don't need to do this now, but it kind of sounds like these are flags instead of a single "kind".

Reply:
Hm, lemme think if I can put these somewhere else 🤔